packages/framework/src/Model/Product/Availability/ProductAvailabilityFacade.php

```diff
@@ -264,6 +276,10 @@ public function getOutOfStockText(string $domainLocale): string
      */
     public function getNotOnStockQuantity(Product $product, int $domainId, int $quantityToAdd): int
     {
+        if ($product->isMainVariant()) {
+            return 0;
```

Comment:
should this method return 0 for main variants? it seems a bit off when this method returns 0 and other methods returning stock quantity return null.

Reply:
true true, I was too lazy to bother with that as this is called in the cart only (never for variants) but yes, it should be in accordance with the other methods. I will fix it